adc/ade7913/ade7913-driver.c

```diff
@@ -63,16 +59,23 @@
 	} while (0)
 
 
+#ifndef ADE7913_PRIO
+#define ADE7913_PRIO 4
+#endif
+
+/* ADE7913_BUF_NUM - number of dma buffers, needs to be power of two */
+#ifndef ADE7913_BUF_NUM
+#define ADE7913_BUF_NUM 4
+#endif
+
 /*
- * The buffer needs to be aligned with:
- * num_of_devices * bytes_per_device * num_of_buffers
- * so that in each filled buffer there will be same amount
- * of samples for each device.
+ * ADE7913_BUF_SIZE - size of contiguous block of memory used by dma,
+ * needs to divisible by (4*16*ADE7913_BUF_NUM) and to be power of two
  */
-#define NUM_OF_BUFFERS  4
-#define ADC_BUFFER_SIZE (4 * NUM_OF_BUFFERS * _PAGE_SIZE)
+#ifndef ADE7913_BUF_SIZE
+#define ADE7913_BUF_SIZE (4 * (ADE7913_BUF_NUM) * (_PAGE_SIZE))
+#endif
```

Comment:
Hmmm, imho because of strict requirements it would be better to fix this one and allow only `ADE7913_BUF_NUM` modification